client/src/components/DogParkShow.js

```diff
@@ -34,9 +41,48 @@ const DogParkShow = (props) => {
         return <p className="tag-cloud-individual-tag">{tag}</p>
     })
 
+    const deleteReview = async (reviewId) => {
+        try {
+            const response = await fetch(`/api/v1/parks/${parkId}/reviews/${reviewId}`, { method: "DELETE"})
+            if(!response.ok) {
+                const errorMessage = `${response.status} (${response.statusText})`
+                const error = new Error(errorMessage)
+                throw error
+            }
+            const newReviews = park.reviews.filter(review => {
+                return (review.id !== reviewId)
+            })
+            setPark({...park, reviews: newReviews })
+        } catch (err) {
+            console.error(`Error in fetch: ${err.message}`)
+        }
+    }
+
+    const handleOnClickDeleteReview = (event, reviewId) => {
+        event.preventDefault()
+        deleteReview(reviewId)
+    }
+
     const reviewsList = park.reviews.map(review => {
+        let userId = null
+        if (props.user) {
+            userId = props.user.id
+        }
+        let deleteButton = ""
+        if (isAdmin || review.userId === userId) {
+            deleteButton =
+                <button
+                    className="button"
+                    onClick={(event) => {
+                    handleOnClickDeleteReview(event, review.id)}}>
+                        Delete review
+                </button>
+        }
```

Comment:
consider moving this logic to the ReviewTile to clean up this DogParkShow component